learning_observer/learning_observer/auth/social_sso.py

```diff
@@ -143,6 +169,20 @@ async def social_handler(request):
     return aiohttp.web.HTTPFound(url)
 
 
+async def handle_lms_request_authorization(request):
+    """
+    Handles the authorization of the specified Learning Management System (LMS)
+    by checking the roster data source and delegating the request to the appropriate handler 
+    based on the data source type.
+    """
+    # Retrieve the data source type for roster data from the PMSS settings
+    roster_source = settings.pmss_settings.source(types=['roster_data'])
+    
+    # Handle the request depending on the roster source
+    if roster_source == 'canvas':
+        await _canvas(request)
```

Comment:
Handlers like this ought to be a dictionary mapping LMS to handling function. We don't want a giant conditional when we eventually integrate 10 more LMSs.

Reply:
Fixed